core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -316,14 +324,18 @@ private void reportMetrics(
       TableIdentifier tableIdentifier,
       MetricsReport report,
       Supplier<Map<String, String>> headers) {
-    reporter.report(report);
     try {
-      client.post(
-          paths.metrics(tableIdentifier),
-          ReportMetricsRequest.of(report),
-          null,
-          headers,
-          ErrorHandlers.defaultErrorHandler());
+      reporter.report(report);
+      if (reportingViaRestEnabled) {
+        client.post(
+            paths.metrics(tableIdentifier),
+            ReportMetricsRequest.of(report),
+            null,
+            headers,
+            ErrorHandlers.defaultErrorHandler());
+      } else {
+        LOG.debug("Reporting metrics to REST endpoint is disabled");
```

Comment:
I don't think this is a helpful debug message every time a report may go out. There isn't anything specific to the report here. I'd rather remove this. If we want to have a debug log for config properties, then we can log when we set `reportingViaRestEnabled`.

Reply:
fair enough, I've removed the debug logging here and rebased the PR